### PR TITLE
Kafka documentation update

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## Description of the change

> Refined the Kafka doc and added information on the serializing data using Avro.

## Type of documentation

- [ ] New documentation
- [x] Documentation update
- [ ] Hotfix

## Notion ticket link

> [Kafka Avro support](https://www.notion.so/rudderstacks/Kafka-Avro-Support-f577b500fcad485099c58113b671b9d2)

### CODE REVIEW

File: docs/destinations/streaming-destinations/kafka.mdx [hunk 1]
```diff
@@ -26,78 +24,123 @@ Before configuring your source and destination on the [dashboard](https://app.ru
...
-- **Username**: Provide the username as configured in Kafka for authenticating clients with SASL.
-- **Password**: Provide the password as configured in Kafka for authenticating clients with SASL.
+   - **CA Certificate**: If SSL is enabled, enter the CA certificate.
+- **Enable SASL with SSL**: If you have enabled SSL, you can use SASL for client authentication by enabling this option.
+   - **SASL Type**: Choose the SASL authentication type from **Plain**, **SCRAM SHA-512**, and **SCRAM SHA-256**.
+   - **Username**: Enter the username configured in Kafka for authenticating clients with SASL.
+   - **Password**: Enter the password configured in Kafka for authenticating clients with SASL.
 
 <div class="warningBlock">
 You need to enable SSL to use SASL authentication.
```
[the remark this answers is not in the document]
Comment (ItsSudip) [reply docs/destinations/streaming-destinations/kafka.mdx:49]: From the webapp, we can do that but that is not the actual flow. I verified from the server code that we are considering `SASL` only when `SSL` is enabled.

**Review by ItsSudip — COMMENTED** [review]
We need to add the config level changes as well. I have added two options in the dashboard viz. `Convert Data To AVRO Format` and `AVRO Schema List`. Using the first option we can enable the serialization and using the second option, we can send a map of schemaId and schema.

**Comment by ameypv-rudder** [comment]
> 

thanks @ItsSudip I will update the connection settings image and add this information.

**Comment by ameypv-rudder** [comment]
@ItsSudip I've incorporated all the suggestions - let me know if the changes look good.

**Review by ItsSudip — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

File: docs/destinations/streaming-destinations/kafka.mdx [hunk 2]
```diff
@@ -26,92 +24,158 @@ Before configuring your source and destination on the [dashboard](https://app.ru
...
+}
+```
 
-<div class="infoBlock">
+The corresponding schema is shown below:
 
-For more information on the Apache Kafka SASL authentication, visit the [official documentation](https://kafka.apache.org/documentation/#security_sasl).
-</div>
+```json
+{"name":"MyClass","type":"record","namespace":"com.acme.avro","fields":[{"name":"anonymousId","type":"string"},{"name":"event","type":"string"},{"name":"messageId","type":"string"},{"name":"originalTimestamp","type":"string"},{"name":"receivedAt","type":"string"},{"name":"request_ip","type":"string"},{"name":"rudderId","type":"string"},{"name":"sentAt","type":"string"},{"name":"timestamp","type":"string"},{"name":"type","type":"string"},{"name":"context","type":"record","fields":[{"name":"traits","type":"record","fields":[{"name":"email","type":"string"}]},{"name":"ip","type":"string"}]},{"name":"integrations","type":"record","fields":[{"name":"KAFKA","type":"record","fields":[{"name":"schemaId","type":"string"}]}]}]}
```
Comment (utsabc) [inline docs/destinations/streaming-destinations/kafka.mdx:93]:
```suggestion
{"name":"RudderStackEvent","type":"record","namespace":"com.acme.avro","fields":[{"name":"anonymousId","type":"string"},{"name":"event","type":"string"},{"name":"messageId","type":"string"},{"name":"originalTimestamp","type":"string"},{"name":"receivedAt","type":"string"},{"name":"request_ip","type":"string"},{"name":"rudderId","type":"string"},{"name":"sentAt","type":"string"},{"name":"timestamp","type":"string"},{"name":"type","type":"string"},{"name":"context","type":"record","fields":[{"name":"traits","type":"record","fields":[{"name":"email","type":"string"}]},{"name":"ip","type":"string"}]},{"name":"integrations","type":"record","fields":[{"name":"KAFKA","type":"record","fields":[{"name":"schemaId","type":"string"}]}]}]}
```

**Review by utsabc — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**